### ISSUE TITLE
completion handler not working right

### ISSUE THREAD

**Issue by odedharth** [hirohisa/ImageLoaderSwift#72]
Hi,

When I load an image to an UIImageView view:

```
            topProductImage.load(leftProduct.imageURL, placeholder: nil) {
                URL, image, error, cacheType in
                if error == nil {
// this part is being called a couple of seconds (around 5-10 seconds) after the image was loaded
                }
            }
```

The completion handler code was being called with a delay of couple of seconds.

**Comment by hirohisa** [comment]
Thank you for the issue, @odedharth .
I will investigate it.

**Comment by hirohisa** [comment]
I improve to callback completionhandler in main thread.
This issue is fixed by version 0.8.8.
